src/unique_arc.rs

```diff
@@ -183,6 +183,33 @@ impl<T> UniqueArc<[MaybeUninit<T>]> {
     }
 }
 
+impl<H, T> UniqueArc<HeaderSlice<H, [MaybeUninit<T>]>> {
+    /// Creates an Arc for a HeaderSlice using the given header struct and allocated space
+    /// for an unitialized slice of length `len`.
+    pub fn from_header_and_uninit_slice(header: H, len: usize) -> Self {
```

Comment:
nit: can new_uninit_slice be updated to use this?

Reply:
It can, but it still requires using the plain Arc temporarily to use it's ` From<Arc<T>> for Arc<HeaderSlice<(), T>> ` implementation. I left the safety comment as it seemed appropriate even w/o an actual safety block.